epictrack-web/src/components/workPlan/status/StatusContext.tsx

```diff
@@ -1,18 +1,146 @@
-import { createContext, useContext } from "react";
-import { Status } from "../../../models/status";
+import { Dispatch, SetStateAction, createContext } from "react";
 import React from "react";
-import { WorkplanContext } from "../WorkPlanContext";
+import TrackDialog from "../../shared/TrackDialog";
+import StatusForm from "./StatusForm";
+import { Status } from "../../../models/status";
+import { showNotification } from "../../shared/notificationProvider";
+import { getAxiosError } from "../../../utils/axiosUtils";
+import { COMMON_ERROR_MESSAGE } from "../../../constants/application-constant";
+import statusService from "../../../services/statusService";
+import workService from "../../../services/workService/workService";
 
-// interface StatusContextProps {}
+interface StatusContextProps {
+  setShowStatusForm: Dispatch<SetStateAction<boolean>>;
+  status?: Status | null;
+  setStatus: Dispatch<SetStateAction<Status | undefined>>;
+  onSave(data: any, callback: () => any): any;
+  setShowApproveStatusDialog: Dispatch<SetStateAction<boolean>>;
+}
 
-// export const StatusContext = createContext<StatusContextProps>({});
+export const StatusContext = createContext<StatusContextProps>({
+  setShowStatusForm: () => ({}),
+  setShowApproveStatusDialog: () => ({}),
+  status: null,
+  setStatus: () => ({}),
+  onSave: (data: any, callback: () => any) => ({}),
+});
 
-// export const StatusProvider = ({
-//   children,
-// }: {
-//   children: JSX.Element | JSX.Element[];
-// }) => {
-//   return (
-//     <StatusContext.Provider value={null}>{children}</StatusContext.Provider>
-//   );
+// const testStatus: Status = {
+//   id: 1,
+//   title: "Toms Status",
+//   description: "this is the description for toms status",
+//   active: true,
+//   high_priority: false,
+//   start_date: "2023-19-26",
+//   approved: true,
 // };
+
+export const StatusProvider = ({
+  children,
+}: {
+  children: JSX.Element | JSX.Element[];
+}) => {
+  const [showStatusForm, setShowStatusForm] = React.useState<boolean>(false);
+  const [showApproveStatusDialog, setShowApproveStatusDialog] =
+    React.useState<boolean>(false);
+  const [status, setStatus] = React.useState<Status>();
+
+  // React.useEffect(() => {
+  //   setStatus(testStatus);
+  // }, []);
+
+  const onDialogClose = () => {
+    setShowStatusForm(false);
+  };
+
+  const onSave = React.useCallback(
```

Comment:
I don't think we need a useCallback here. onSave could be just a simple arrow function